lib/apiservers/engine/backends/container.go

```diff
@@ -118,43 +143,18 @@ func (c *Container) ContainerCreate(config types.ContainerCreateConfig) (types.C
 				http.StatusInternalServerError)
 	}
 
-	c.m.Lock()
-	defer c.m.Unlock()
-
-	layer, found := c.HackMap[config.Config.Image]
-	if !found {
-		// FIXME: This is a temporary workaround until we have a name resolution story.
-		// Call imagec with -resolv parameter to learn the name of the vmdk and put it into in-memory map
-		cmdArgs := []string{"-reference", config.Config.Image, "-resolv", "-standalone", "-destination", os.TempDir()}
+	var layer *viccontainer.VicContainer
+	container := viccontainer.GetCache().GetContainerByName(config.Config.Image)
+	if container == nil {
+		layer, err = c.getImageMetadataFromStoragePL(config.Config.Image)
 
-		out, err := goexec.Command(imagec, cmdArgs...).Output()
 		if err != nil {
-			log.Printf("%s exit code: %s", imagec, err)
-			return types.ContainerCreateResponse{},
-				derr.NewErrorWithStatusCode(fmt.Errorf("Container look up failed"),
-					http.StatusInternalServerError)
-		}
-		var v1 struct {
-			ID string `json:"id"`
-			// https://github.com/docker/engine-api/blob/master/types/container/config.go
-			Config container.Config `json:"config"`
-		}
-		if err := json.Unmarshal(out, &v1); err != nil {
-			return types.ContainerCreateResponse{},
-				derr.NewErrorWithStatusCode(fmt.Errorf("Failed to unmarshall image history: %s", err),
-					http.StatusInternalServerError)
+			return types.ContainerCreateResponse{}, err
```

Comment:
Not really, I briefly looked at what docker does. They do ad-hoc error reporting without using any libraries or pkgs so I ended up doing something similar in imagec. 

I think error reporting should be an epic that needs to be worked on.